### ISSUE TITLE
[Details] Putting interactive components in Summary creates WCAG issue

### ISSUE THREAD

**Issue by kklein-celera** [vaadin/flow-components#7674]
### Description

The documentation for the Details component says that any component can be added to the Summary. However, if you add an interactive component (e.g. an Anchor), this generates a WCAG violation, because the interactive component is nested within the interactive Summary itself.
See https://dequeuniversity.com/rules/axe/4.10/nested-interactive?application=AxeChrome
Is there a way to restructure the Summary so that additional components are not nested?

### Expected outcome

The component should generate HTML that is compliant with WCAG standards.

### Minimal reproducible example

`Anchor anchor1 = new Anchor(url, "Test");
 Details itemDetails = new Details(anchor, new VerticalLayout(new Text("Hello, World")));
`

### Steps to reproduce

Create a Details component and instead of putting Text in the Summary, put in an Anchor component.
Then check the resulting page with a WCAG checker (I use the Axe DevTools extension for Chrome). Note the violation generated for nested interactive components.

### Environment

Vaadin version(s): 24.7.8
OS: Windows

### Browsers

Chrome, Issue is not browser related

**Comment by rolfsmeds** [comment]
Indeed, I don't think there is any way to "fix" this in the component, so we will need to add a disclaimer about this in the documentation.
